### PR TITLE
Change exit code to no success when infeasible

### PR DESCRIPTION
**Pull request by timtroendle**
This is a breaking change.
This fixes https://github.com/calliope-project/calliope/issues/182.

Fixes issue(s) #182 

Reviewer checklist:

- [x] Test(s) added to cover contribution
- [ ] ~~Documentation updated~~
- [x] Changelog updated
- [x] Coverage maintained or improved

### CODE REVIEW

File: calliope/cli.py [hunk 1]
```diff
@@ -259,7 +268,10 @@ def run(model_file, scenario, save_netcdf, save_csv, save_plots,
                     'Model termination condition non-optimal. Not saving plots',
                     fg='red', bold=True
                 )
-        print_end_time(start_time)
+        if fail_when_infeasible and termination != 'optimal':
+            raise BackendError("Problem is infeasible.")
```
Comment (timtroendle) [inline calliope/cli.py]: Not sure whether this is the right error class.

File: calliope/cli.py [hunk 2]
```diff
@@ -56,6 +57,13 @@
     help='Filename to save profile to if enabled --profile.'
 )
 
+
+_fail_when_infeasible = click.option(
+    '--fail_when_infeasible/--no_fail_when_infeasible', is_flag=True, default=True,
```
Comment (timtroendle) [inline calliope/cli.py]: I followed Calliope's convention of using underscores in command line options. But is that something you consider changing at one point, maybe for a 1.0 release? ;-).

Comment (brynpickering) [reply calliope/cli.py]: I think this could be updated a bit earlier than a 1.0 release ;)

File: calliope/test/test_cli.py [hunk 3]
```diff
@@ -120,3 +123,22 @@ def test_generate_scenarios(self):
...
+            "--override_dict={techs.test_supply_elec.constraints.energy_cap_max: 1}"  # elec supply too low
+        ])
+        assert result.exit_code != 0
+
+    def test_success_exit_code_when_infeasible_and_demanded(self):
+        runner = CliRunner()
+        result = runner.invoke(cli.run, [
+            _MINIMAL_TEST_MODEL,
+            "--no_fail_when_infeasible",
+            "--scenario=supply_purchase,investment_costs",  # without these, the model cannot run
```
Comment (brynpickering) [inline calliope/test/test_cli.py]: supply_purchase means that it is running a MILP model... I would go with 'supply_simple' to have the smallest possible model running (if that allows the model to run, ofc)

File: changelog.rst [hunk 4]
```diff
@@ -8,6 +8,8 @@ Release History
 
 |new| Documentation for developers has been improved to include an overview of the internal package structure and a guide to contributing code via a pull request.
 
+|changed| Exit code for infeasible problems now is 1 (no sucess). This is a breaking change when relying on the exit code.
```
Comment (brynpickering) [inline changelog.rst]: success*

File: calliope/test/test_cli.py (hunk 3 above)
Comment (timtroendle) [reply calliope/test/test_cli.py]: `supply_simple` doesn't exist but is not necessary anyway: `investment_costs` alone has all we need.

**Pull request merged**